test/Swap.test.ts

```diff
@@ -221,5 +221,13 @@ describe("Swap RBTC", function () {
     expect(senderBalance).to.be.equal(oneEther);
 
     await expect(anotherSideToken.connect(sender).send(swapRBTC.address, halfEther, "0x")).to.be.revertedWith("SwapRBTC: Side Token not found");
-41  });
+  });
+
+  it('Should validate an invalid address when call the token received method', async function () {
+    await expect(swapRBTC.tokensReceived(bnbContract, sender.address, deployer.address, 0, 0, 0)).to.be.revertedWith("SwapRBTC: Invalid 'to' address");
+  });
+
+  it('Should validate if the sender address is already included in the list on token received method', async function () {
+    await expect(swapRBTC.tokensReceived(deployer.address, sender.address, swapRBTC.address, 0, 0, 0)).to.be.revertedWith("SwapRBTC: Side Token not found");
```

Comment:
Here we are not validating the sender address, we are validating that the token is listed. The sender address should be the token address, try to use names that represent this. it should look something like:
`swapRBTC.connect(token).tokensReceived(deployer.address, sender.address, swapRBTC.address, 0, 0, 0)`

This tests are ok to test specifics, of the method, but to test the actual hook a test token of type ERC777 should be created and then a user would have to call the .send method of that token to test the actual hook being called as it's supposed to https://eips.ethereum.org/EIPS/eip-777#sending-tokens